colcon_package_selection/package_selection/dependencies.py

```diff
@@ -77,6 +77,8 @@ def select_packages(self, args, decorators):  # noqa: D102
         if args.packages_select_by_dep:
             deps = set(args.packages_select_by_dep)
             for decorator in decorators:
+                if decorator.descriptor.name in deps:
+                    decorator.selected = False
```

Comment:
Please modify the condition in the next line instead. That will avoid the duplicate / inverse logic as well as print the log message for the skipped package names.